unumpy/multimethods.py

```diff
@@ -506,56 +495,49 @@ def broadcast_arrays(*args, subok=False):
     return args
 
 
-@create_numpy(_self_argreplacer)
+@create_numpy(_generic_argreplacer('array'))
 @all_of_type(ndarray)
 def broadcast_to(array, shape, subok=False):
     return (array,)
 
 
-def _first_argreplacer(args, kwargs, arrays1):
-    def func(arrays, axis=0, out=None):
-        return (arrays1,), dict(axis=0, out=None)
-
-    return func(*args, **kwargs)
-
-
-@create_numpy(_first_argreplacer)
+@create_numpy(_generic_argreplacer('arrays'))
 @all_of_type(ndarray)
 def concatenate(arrays, axis=0, out=None):
     return arrays
 
 
-@create_numpy(_first_argreplacer)
+@create_numpy(_generic_argreplacer('arrays'))
```

Comment:
Here, you're replacing the whole argument with one array. You must replace `arrays` with a list of all passed-in arrays.